tests/phpunit/integration/Modules/AnalyticsTest.php

```diff
@@ -153,6 +154,63 @@ public function test_get_datapoints() {
 		);
 	}
 
+	/**
+	 * @dataProvider data_parse_date_range
+	 */
+	public function test_parse_date_range( $period_requested, $previous_period_end_offset ) {
```

Comment:
Since the method belongs to the base `Module` and has not changed in the Analytics class (only how it's called), this test should move to `\Google\Site_Kit\Tests\Core\Modules\ModuleTest` respectively and use `FakeModule` for the instance. 

By doing this we can also add `parse_date_range` to that class to make it public so that using `Reflection` isn't necessary here. (I know that was my suggestion but I must have missed that there wasn't anything Analytics-specific in the implementation at the time - or it changed 😄 ). See `FakeModule::exception_to_error`.

Reply:
Ok, I'll move this. Thanks for the tip.

Reply:
I moved the test to `tests/phpunit/integration/Core/Modules/ModuleTest.php` - that makes more sense.

> By doing this we can also add parse_date_range to that class to make it public so that using Reflection isn't necessary here.

@aaemnnosttv - By "that class" do you mean `Google\Site_Kit\Core\Modules`? That is where `parse_date_range` already is. So, should I change the visibility to public and remove the reflection in the test? I feel like these helpers should be private/protected unless we have a reason to expose them - what do you think @felixarntz?

Reply:
No sorry, I meant you can define the same method on `FakeModule` as `public` and simply call `parent::parse_date_range` with the same args so that it's callable without reflection for the test but doesn't need to be in the base `Module` class. Like this method https://github.com/google/site-kit-wp/blob/d68b6cedfcef57cb8cdb44e02dc00ca51b590182/tests/phpunit/includes/Core/Modules/FakeModule.php#L203-L205

Reply:
Got it, fixed in [`476fd98` (#1822)](https://github.com/google/site-kit-wp/pull/1822/commits/476fd98d9ed7ad15f0d7d972a3a82d83bb875b98)